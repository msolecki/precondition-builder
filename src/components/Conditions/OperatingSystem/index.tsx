import React, {ChangeEvent} from 'react'
import {makeStyles} from '@material-ui/core/styles'
import MenuItem from '@material-ui/core/MenuItem'
import FormControl from '@material-ui/core/FormControl'
import Select from '@material-ui/core/Select'
import PropTypes from 'prop-types'
import Title from '../../Title'

const useStyles = makeStyles(({spacing}) => ({
    root: {
        display: 'flex',
        flexWrap: 'wrap',
    },
    formControl: {
        marginTop: spacing(1),
        marginBottom: spacing(3),
        minWidth: 120,
    },
}))

export interface OperatingSystemProps {
    onDataChange(value: string | null): void;

    system: string | null;
}

const OperatingSystem: React.FC<OperatingSystemProps> = (props: OperatingSystemProps): React.ReactElement => {
    const classes = useStyles()

    function handleChange(event: ChangeEvent<{}>): void {
        const {value} = event.target as HTMLSelectElement

        props.onDataChange(value)
    }

    return (
        <>
            <Title text="Operating system"/>
            <FormControl className={classes.formControl} fullWidth required component='div'>
                <Select
                    value={props.system === null ? '' : props.system}
                    displayEmpty
                    onChange={handleChange}
                    inputProps={{
                        name: 'system',
                        id: 'system',
                    }}
                >
                    <MenuItem component='li' button={true} value={'iOS'}>iOS</MenuItem>
                    <MenuItem component='li' button={true} value={'Android'}>Android</MenuItem>
                </Select>
            </FormControl>
        </>
    )
}

OperatingSystem.propTypes = {
    onDataChange: PropTypes.func.isRequired,
    system: PropTypes.string,
}

OperatingSystem.defaultProps = {
    system: null,
}

export default OperatingSystem
