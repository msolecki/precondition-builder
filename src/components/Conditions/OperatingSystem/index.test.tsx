import * as React from 'react'
import {ReactTestRenderer} from 'react-test-renderer'
import {createShallow} from '@material-ui/core/test-utils'

import OperatingSystem, {OperatingSystemProps} from './index'
import {Select} from '@material-ui/core'

const component = (props: OperatingSystemProps): OperatingSystem => <OperatingSystem {...props}/>
const onDataChangeFunc = jest.fn()

describe('<OperatingSystem />', () => {
    let shallow

    beforeAll(() => {
        shallow = createShallow()
    })

    test('empty snapshot', () => {
        const wrapper: ReactTestRenderer = shallow(
            component({
                system: null,
                onDataChange: () => {
                }
            }))

        expect(wrapper).toMatchSnapshot()
    })

    test('default snapshot', () => {
        const wrapper = shallow(
            component({
                system: 'ios',
                onDataChange: onDataChangeFunc
            }))
        const event = {target: {name: 'name', value: 'value'}}

        wrapper.find(Select).simulate('change', event)

        expect(onDataChangeFunc).toHaveBeenCalledWith('value')
    })
})
