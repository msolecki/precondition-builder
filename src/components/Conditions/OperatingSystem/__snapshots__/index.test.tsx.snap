// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OperatingSystem /> empty snapshot 1`] = `
<Fragment>
  <Title
    text="Operating system"
  />
  <WithStyles(ForwardRef(FormControl))
    className="makeStyles-formControl-2"
    component="div"
    fullWidth={true}
    required={true}
  >
    <WithStyles(WithFormControlContext(ForwardRef(Select)))
      displayEmpty={true}
      inputProps={
        Object {
          "id": "system",
          "name": "system",
        }
      }
      onChange={[Function]}
      value=""
    >
      <WithStyles(ForwardRef(MenuItem))
        button={true}
        component="li"
        value="iOS"
      >
        iOS
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        button={true}
        component="li"
        value="Android"
      >
        Android
      </WithStyles(ForwardRef(MenuItem))>
    </WithStyles(WithFormControlContext(ForwardRef(Select)))>
  </WithStyles(ForwardRef(FormControl))>
</Fragment>
`;
